File: javascript/bob/bob.js
var Bob = function(){
  this.hey = function(msg) {
    greetings = 'Whatever.';

    if ( msg.match(/^([A-Z] !?)*$/) > 0 ){
      greetings  ='Woah, chill out!';
    }

    if (msg == 'Does this cryogenic chamber make me look fat?'){
      greetings  ='Sure.';
    }

    return greetings
  }
}
module.exports = Bob;

File: clojure/bob/bob.clj
(ns bob
  (:require [clojure.string :refer [blank? upper-case]]))

(defn screaming? [msg]
  (and
    (= (upper-case msg) msg)
    (not (blank? msg))
    (re-find #"[A-Z]+" msg)))

(defn asking? [msg]
  (re-find #"\?$" msg))

(defn silence? [msg]
  (re-find #"^\s*$" msg))


(defn response-for [msg]
  (cond
    (screaming? msg) "Woah, chill out!"
    (asking? msg) "Sure."
    (silence? msg) "Fine. Be that way!"
    :else "Whatever."))
